src/transformers/trainer_utils.py

```diff
@@ -653,7 +653,21 @@ def find_executable_batch_size(
 
         return accelerate_find_executable_batch_size(function=function, starting_batch_size=starting_batch_size)
 
-    return functools.partial(function, batch_size=starting_batch_size)
+    batch_size = starting_batch_size
+    while True:
+        try:
+            # Call the function with the current batch size
+            result = function(batch_size)
+            # Check if the learning rate scheduler needs to be updated based on the current batch size
+            if hasattr(result, 'lr_schedulers'):
+                for scheduler in result.lr_schedulers:
+                    if hasattr(scheduler, 'num_batches'):
+                        scheduler.num_batches = math.ceil(result.num_train_examples / batch_size)
+            return result
+        except (RuntimeError, OverflowError, ValueError):
+            # Cut the batch size in half and try again
+            batch_size //= 2
+
```

Comment:
Don't we need to break here if `batch_size` becomes 0?

Reply:
@sgugger You are right. revised it.

Reply:
Actually, isn't this whole if else block redundant? And we're reducing the bs by 2^2 here, Accelerate needs to handle this. This PR should just touch the learning rate scheduler: https://github.com/huggingface/accelerate/blob/main/src/accelerate/utils/memory.py#L128-L142

Reply:
Not sure if the mentioned snippet would actually cover everything but I guess in that case, if Accelerate should handle this then we can close this PR.

Reply:
Yes this is something Accelerate should handle, as the find_executable_batch_size function only should work if `auto_find_batch_size=True`. This was a good start, and the right idea, but it needs to be over there I'm afraid.

Reply:
(This wouldn't have been the case without the fact that the Trainer in torch runs on Accelerate now)

Reply:
@muellerzr Please take look at this PR [#1632](https://github.com/huggingface/accelerate/pull/1632) and see if it makes sense for you.

Reply:
@muellerzr I close this PR and take look at [#1633](https://github.com/huggingface/accelerate/pull/1633) PR. if it makes sense. if not you can close that one too. Thank you!